plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -831,6 +831,40 @@ private void deleteModel(String modelId) {
             .setIndicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN)
             .setAbortOnVersionConflict(false);
         client.execute(DeleteByQueryAction.INSTANCE, deleteChunksRequest);
+        // This checks if model group is created when registering the version. If yes, model group is deleted since the version registration
+        // had
+        // failed. Else model group latest version is decremented by 1
+        deleteOrUpdateModelGroup(registerModelInput.getModelGroupId(), registerModelInput.getDoesVersionCreateModelGroup(), modelVersion);
+    }
+
+    private void deleteOrUpdateModelGroup(String modelGroupID, Boolean doesVersionCreateModelGroup, String modelVersion) {
+        if (doesVersionCreateModelGroup) {
+            DeleteRequest deleteModelGroupRequest = new DeleteRequest();
+            deleteModelGroupRequest.index(ML_MODEL_GROUP_INDEX).id(modelGroupID).setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+            client.delete(deleteModelGroupRequest);
+        } else {
+            updateLatestVersionInModelGroup(
+                modelGroupID,
+                modelVersion,
+                ActionListener
+                    .wrap(r -> log.debug("model group updated, response {}", r), e -> log.error("Failed to update model group", e))
+            );
+        }
+    }
+
+    private void updateLatestVersionInModelGroup(String modelGroupID, String modelVersion, ActionListener<UpdateResponse> listener) {
```

Comment:
Is it possible to make this method more generic to update the model version in both ways. (increment and decrement both)?